mrp_mto_with_stock/models/mrp_production.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2017-18 Eficent Business and IT Consulting Services S.L.
+# Copyright 2015 John Walsh
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.exceptions import UserError
+import copy
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    def _mto_with_stock_condition(self, move):
+        """Extensibility-enhancer method for modifying the scenarios when
+        MTO/MTS method should apply."""
+        return move.location_id in move.product_id.mrp_mts_mto_location_ids
+
+    @api.multi
+    def action_assign(self):
+        """Reserves available products to the production order but also creates
+        procurements for more items if we cannot reserve enough (MTO with
+        stock).
+        @returns True"""
+        # reserve all that is available (standard behaviour):
+        res = super(MrpProduction, self).action_assign()
+        # try to create procurements:
+        move_obj = self.env['stock.move']
+        for production in self:
+            warehouse = production.location_src_id.get_warehouse()
+            mto_with_no_move_dest_id = warehouse.mrp_mto_mts_forecast_qty
+            move_ids = copy.copy(production.move_raw_ids.ids)
+            for move in move_obj.browse(move_ids):
```

Comment:
Why not using recordsets directly ?

Reply:
it comes from [this commit](https://github.com/OCA/manufacture/commit/f206743efef91a4c61b7540c3c2b828113c5ada6), I'm unable to find some conversation about